lib/server.rb

```diff
@@ -328,7 +330,13 @@ def diff_to_object(diff)
   end
 
   get "/:id" do
-    @description = SystemDescription.load(params[:id], settings.system_description_store)
+    begin
+      @description = SystemDescription.load(params[:id], settings.system_description_store)
+    rescue Machinery::Errors::SystemDescriptionNotFound => e
+      session[:error] = e.to_s
+      status 404
```

Comment:
Thanks for reviewing. 

I wondered the same thing. The final response will be a 302 anyway. I have removed it. :)